concordium-node/src/configuration.rs

```diff
@@ -881,36 +881,34 @@ pub struct AppPreferences {
 
 impl AppPreferences {
     /// Creates an `AppPreferences` object.
-    pub fn new(override_conf: PathBuf, override_data: PathBuf) -> Self {
+    pub fn new(override_conf: PathBuf, override_data: PathBuf) -> anyhow::Result<Self> {
         let file_path = Self::calculate_config_file_path(&override_conf, APP_PREFERENCES_MAIN);
-        let mut new_prefs = match OpenOptions::new().read(true).write(true).open(&file_path) {
-            Ok(file) => {
-                let mut reader = BufReader::new(&file);
-                let load_result = PreferencesMap::<String>::load_from(&mut reader);
-                let prefs = load_result.unwrap_or_else(|_| PreferencesMap::<String>::new());
-
-                AppPreferences {
-                    preferences_map:     prefs,
-                    override_data_dir:   override_data,
-                    override_config_dir: override_conf,
-                }
+        let mut new_prefs = if let Ok(file) = OpenOptions::new().open(&file_path) {
```

Comment:
This is pedantic, but the logic here is ugly since we are checking whether a file exists by trying to open it.

Instead, a cleaner logic would be something like (see https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file )
```
if file_path.is_file() {
   let file = std::fs::File::open(&file_path).context("Could not open configuration file.")?;
   ...
} else {
  // the current else branch
}
```

Reply:
Agree, we should take the time now to clean up this logic. 
But other than that it looks good! 👍

Reply:
Got caught up in something else yesterday, taking a look at this now!

Reply:
Tried doing this.